chimedb/dataflag/client.py

```diff
@@ -144,11 +182,386 @@ def convert(self, value, param, ctx):
 # =========================
 
 
+@db.atomic(read_write=True)
 @click.group()
 def cli():
     """CHIME data flagging tool."""
 
 
+@cli.group("opinion_type")
+def opinion_type():
+    """View and modify data flagging opinion types."""
+    pass
+
+
+@opinion_type.command("list")
+def opinion_type_list():
+    """List known flagging opinion types."""
+    for type_ in DataFlagOpinionType.select():
+        click.echo(type_.name)
+
+
+@opinion_type.command("create")
+@click.argument("name")
+@click.option(
+    "--description", help="Description of flagging opinion type.", default=None
+)
+@click.option(
+    "--metadata", type=JSON, help="Extra metadata as as JSON dict.", default=None
+)
+@click.option("--force", "-f", is_flag=True, help="Create without prompting.")
+def create_opinion_type(name, description, metadata, force):
+    """Create a new data flag opinion type with given NAME, and optional description and metadata.
+    """
+    type_ = DataFlagOpinionType()
+
+    type_.name = name
+    type_.description = description
+    type_.metadata = metadata
+
+    if force:
+        type_.save()
+    else:
+        click.echo("Type to create:\n")
+        click.echo(format_type(type_))
+        if click.confirm("Create type?"):
+            type_.save()
+            click.echo("Success.")
+        else:
+            click.echo("Aborted.")
+
+
+@opinion_type.command("show")
+@click.argument("opinion_type", type=OTYPE, metavar="TYPE")
+def show_opinion_type(opinion_type):
+    """Show details of the specified opinion TYPE."""
+    click.echo(format_type(opinion_type))
+
+
+@cli.group("opinion")
+def opinion():
+    """Insert data flagging opinions."""
+    pass
+
+
+@opinion.command("create")
+@click.argument("type_", type=OTYPE, metavar="TYPE")
+@click.argument("start", type=TIME)
+@click.argument("finish", type=TIME)
+@click.argument(
+    "decision",
+    type=click.Choice(DataFlagOpinion.choices_decision, case_sensitive=False),
+)
+@click.option("--description", help="Description of flag.", default=None)
+@click.option("--user", "-u", help="Wiki user adding the flag opinion.", required=True)
+@click.option("--password", "-p", help="Wiki user password.", required=True)
```

Comment:
For the CLI, why do we need to authenticate? I don't see a serious need, it just seems like it's going to be annoying.

For the web front end, sure, we want so authentication to stop Joe Haxor coming and messing stuff up, but if I can actually use the CLI I already have direct access to the database, and could do whatever I wanted.

I can understand wanting to validate that the username matches a known one. But even that you could probably just pull from the local username if it's not explicitly supplied. The password seems like it's going to be annoying overkill. In my case, my wiki password (as all of may passwords are) are long random strings that I keep in a password manager, if I have to start copy and pasting that into the command line all the time, I'm going to get frustrated, and it'll discourage using the CLI.

Reply:
Yeah, I agree.

It's basically the way we run the layout database: if you use the web interface, you log in.  If you use the Python code in ch_util, you don't have to log in, but if you want to modify something with the CLI, you do have to provide a username; not as a credentials check, so no password verification, just so it has something to put in the "modified-by" field.

Reply:
From a security perspective: unnecessarily throwing around passwords can only weaken security.